test/project_types/script/layers/infrastructure/languages/typescript_task_runner_test.rb

```diff
@@ -187,6 +207,97 @@
+
+    describe "when npm is not installed" do
+      it "should raise error" do
+        ctx.expects(:capture2e)
+          .with("npm", "--version")
+          .returns([nil, mock(success?: false)])
+        assert_raises Script::Layers::Infrastructure::Errors::NoDependencyInstalledError do
+          subject
+        end
+      end
+    end
+
+    describe "when node is not installed" do
+      it "should raise error" do
+        ctx.expects(:capture2e)
+          .with("npm", "--version")
+          .returns([TS_EXACT_NODE_VERSION, mock(success?: true)])
+        ctx.expects(:capture2e)
+          .with("node", "--version")
+          .returns([nil, mock(success?: false)])
+        assert_raises Script::Layers::Infrastructure::Errors::NoDependencyInstalledError do
+          subject
+        end
+      end
+    end
+
+    describe "when npm version is below minimum" do
+      it "should raise error" do
+        ctx.expects(:capture2e)
+          .with("npm", "--version")
+          .returns([TS_BELOW_NPM_VERSION, mock(success?: true)])
+
+        assert_raises Script::Layers::Infrastructure::Errors::MissingDependencyVersionError do
+          subject
+        end
+      end
+    end
+
+    describe "when npm version is above minimum" do
+      describe "when node version is below minimum" do
+        it "should raise error" do
+          stub_tool_versions(npm: TS_ABOVE_NPM_VERSION, node: TS_BELOW_NODE_VERSION)
+          assert_raises Script::Layers::Infrastructure::Errors::MissingDependencyVersionError do
+            subject
+          end
+        end
+      end
+
+      describe "when node version is above minimum" do
+        it "should install successfully" do
+          stub_tool_versions(npm: TS_ABOVE_NPM_VERSION, node: TS_ABOVE_NODE_VERSION)
+          subject
+        end
+      end
+
+      describe "when node version is the exact version" do
+        it "should install successfully" do
+          stub_tool_versions(npm: TS_ABOVE_NPM_VERSION, node: TS_EXACT_NODE_VERSION)
+          subject
+        end
+      end
+    end
+
+    describe "when npm version is exactly the version" do
+      describe "when node version is below minimum" do
+        it "should raise error" do
+          stub_tool_versions(npm: TS_EXACT_NPM_VERSION, node: TS_BELOW_NODE_VERSION)
+          assert_raises Script::Layers::Infrastructure::Errors::MissingDependencyVersionError do
+            subject
+          end
+        end
+      end
+
+      describe "when node version is above minimum" do
+        it "should install successfully" do
+          stub_tool_versions(npm: TS_EXACT_NPM_VERSION, node: TS_ABOVE_NODE_VERSION)
+          subject
+        end
+      end
+
+      describe "when node version is the exact version" do
+        it "should install successfully" do
+          stub_tool_versions(npm: TS_EXACT_NPM_VERSION, node: TS_EXACT_NODE_VERSION)
+          subject
+        end
+      end
+    end
+  end
```

Comment:
The tests don't need to be written this way in subclasses any more.  We should focus on the contract between the subclass and its super class.  For example, the test from 285-290 and 292-296 are identical.  Basically, what you need to test here is that the subclass is calling `TaskRunner::check_tool_version!` with the right input, and processing the output correctly.